File: src/ruby/NinjaSurvivalRate/src/skill.rb
class Skill
  
  def initialize (name)
    @name=name
  end
  
  def available_to (opponent)
    if @name=="roundhouse-kick" then
      return opponent=="Chuck Norris"
    end
    true
  end
  
  def danger (opponent)
    if opponent=="Chuck Norris" then
       "extreme"
  	elsif opponent=="samurai" and @name=="katana" then
      "high"
  	else
      "low"
  	end
  end
  def name
    @name
  end
end

File: src/dotnet/NinjaSurvivalRateWithSharedState/src/Skill.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace NinjaSurvivalRate
{
    class Skill
    {
        private String name;
	    public Skill(String name){
		    this.name=name;
	    }
	    public bool AvailableTo(String opponent){
		    if ("Chuck Norris" == opponent) return true;
		    if ("roundhouse-kick" == name) return false;
		    return true;		
	    }
	    public String Danger(String opponent){
		    if ("Chuck Norris" == opponent) return "extreme";
		    if ("samurai" == opponent && "katana" == name) return "high";
		    return "low";
	    }
	    public String Name(){
		    return name;
	    }
    }

}
